### PR TITLE
Yoselyn (Completed 3 exercises and HackYourTemperature I)

### PR DESCRIPTION
**Pull request by callejas1**

### CODE REVIEW

File: week1/homework/exercises/1-pad-numbers/script.js [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
+ *
  * Tips:
  *   where to use `exports` and where `require`?
  */
 
-let numbers = [ "12", "846", "2", "1236" ];
+let numbers = ['12', '846', '2', '1236'];
+
+// YOUR CODE GOES HERE
+const padLeft = require('./padLeft'); // import function from padLeft.js
```
Comment (olcarmontzaragoza) [inline week1/homework/exercises/1-pad-numbers/script.js:21]: Perfect import statement ;)

File: week1/homework/exercises/1-pad-numbers/script.js [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
 
-// YOUR CODE GOES HERE
\ No newline at end of file
+// Add space before each value until a total of 4 spaces are taken
+numbers.forEach((value) => {
```
Comment (olcarmontzaragoza) [inline week1/homework/exercises/1-pad-numbers/script.js:24]: Great use of forEach here Yoselyn! You clearly know where to use map() and forEach(), where forEach() is used where you don't need to return anything from the function.

File: week1/homework/exercises/1-pad-numbers/script.js [hunk 3]
```diff
@@ -1,21 +1,27 @@
...
+  const padded = padLeft(value, 4, ' ');
```
Comment (olcarmontzaragoza) [inline week1/homework/exercises/1-pad-numbers/script.js:25]: This is just from a conciseness perspective (my own opinion), but it would be great to combine the function call and console log in as little lines as possible:

console.log ( padLeft(value, 4, '"') );

Otherwise, great job!

File: week1/homework/exercises/2-left-pad/script.js [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
+
+// YOUR CODE GOES HERE
+const leftPad = require('left-pad'); // import from npm
 
-// YOUR CODE GOES HERE
\ No newline at end of file
+// Add space to pad numbers for a total of 8 characters
+numbers.forEach((val) => {
+  const paddedLeft = leftPad(val, 8, ' ');
+  console.log(paddedLeft);
```
Comment (olcarmontzaragoza) [inline week1/homework/exercises/2-left-pad/script.js:19]: Great work!

File: week1/homework/exercises/3-web-server/index.html [hunk 5]
```diff
@@ -1,6 +1,7 @@
 <html>
 	<head>
-		<title>My First Web Server</title>
+    <title>My First Web Server</title>
+    <link rel="stylesheet" type="text/css" href="style.css" />
 	</head>
 	<body>
 		<h1>Hello, anyone there?</h1>
```
Comment (olcarmontzaragoza) [inline week1/homework/exercises/3-web-server/index.html:7]: Haha I like that: "anyone there?"

File: week1/homework/exercises/3-web-server/server.js [hunk 6]
```diff
@@ -2,13 +2,42 @@
...
+      }
+    });
+  }
 });
 
-server.listen(3000); // The server starts to listen on port 3000
\ No newline at end of file
+const PORT = process.env.PORT || 3000; // will look for environment, if not it'll be 3000
+
+server.listen(PORT, () => console.log(`Server running. Port: ${PORT}`)); // The server starts to listen on port 3000
```
Comment (olcarmontzaragoza) [inline week1/homework/exercises/3-web-server/server.js:43]: Very concise, good job! Great use of the arrow function.

File: week1/homework/exercises/3-web-server/server.js [hunk 7]
```diff
@@ -2,13 +2,42 @@
...
-
+const http = require('http');
+const fs = require('fs');
+const path = require('path');
 //create a server
 let server = http.createServer(function (req, res) {
   // YOUR CODE GOES IN HERE
-	res.write('Hello World!'); // Sends a response back to the client
-	res.end(); // Ends the response
+  if (req.url === '/') {
```
Comment (olcarmontzaragoza) [inline week1/homework/exercises/3-web-server/server.js:11]: Hey Yoselyn, the functionality appears to be in intact, but I found this a little hard to follow. You seem to be trying to do too many things in one function that would be better suited to have their separate own functions. Also notice how many of the statements are repeated such as "throw error" or "res.end(...)"- this should be a clue that you need to make it more concise. I want to give you give this another try, but here are some tips:

1- Make a function that sets the content-type by assigning a variable such as var contentType to "text/css", "text/javascript" etc based on the input ("/index.js", "/style.css" etc) (Make this a switch statement if possible!)

2- Once you've done that you can have just one fs.readFile() where you pass in the contentType you found above (and return the data of each)

Good luck!

File: week1/homework/exercises/3-web-server/server.js [hunk 8]
```diff
@@ -2,13 +2,42 @@
...
+    fs.readFile(path.join(__dirname, '', 'index.js'), (error, content) => {
+      if (error) {
+        throw error;
+      } else {
+        res.setHeader('Content-Type', 'text/javascript'); // Send js file
+        res.end(content); // Ends the response
+      }
+    });
+  } else if (req.url === '/style.css') {
+    fs.readFile(path.join(__dirname, '', 'style.css'), (e, style) => {
```
Comment (olcarmontzaragoza) [inline week1/homework/exercises/3-web-server/server.js:30]: Quick tip: You don't necessarily need a new parameter name in each function ("style", "content", "data") as function scope means that they won't interfere with one another.

File: week1/homework/exercises/3-web-server/server.js [hunk 9]
```diff
@@ -2,13 +2,42 @@
  * Exercise 3: Create an HTTP web server
  */
 
-var http = require('http');
-
+const http = require('http');
+const fs = require('fs');
+const path = require('path');
 //create a server
 let server = http.createServer(function (req, res) {
```
Comment (olcarmontzaragoza) [inline week1/homework/exercises/3-web-server/server.js:9]: You may want to use a "const" instead of "var" as you won't be changing the server.

**Review by olcarmontzaragoza — COMMENTED** [review]
Hey Yoselyn, great work on your first week of NodeJS! I'd just recommend you take a look at my comments on "web-server" to keep conciseness, but overall everything appears to be in place, so keep up the great work!

File: week1/homework/exercises/1-pad-numbers/script.js (hunk 2 above)
Comment (callejas1) [reply week1/homework/exercises/1-pad-numbers/script.js:24]: Thank you for your feedback, it's exciting the see this other side of programming

File: week1/homework/exercises/1-pad-numbers/script.js (hunk 3 above)
Comment (callejas1) [reply week1/homework/exercises/1-pad-numbers/script.js:25]: Thank you for your input, Oliver 😸

File: week1/homework/exercises/2-left-pad/script.js (hunk 4 above)
Comment (callejas1) [reply week1/homework/exercises/2-left-pad/script.js:19]: Thanks!

File: week1/homework/exercises/3-web-server/index.html (hunk 5 above)
Comment (callejas1) [reply week1/homework/exercises/3-web-server/index.html:7]: I wanted to go for spooky hahaha funny

File: week1/homework/exercises/3-web-server/server.js (hunk 7 above)
Comment (callejas1) [reply week1/homework/exercises/3-web-server/server.js]: You got it! I pushed a new commit just now, please check if everything works as expected on your end. I'm excited about this one, really great feedback, Oliver. I love how easier to read it looks now, so clean, reusable, simple!

File: week1/homework/exercises/3-web-server/server.js (hunk 8 above)
Comment (callejas1) [reply week1/homework/exercises/3-web-server/server.js]: Oh yeah!!

File: week1/homework/exercises/3-web-server/server.js (hunk 9 above)
Comment (callejas1) [reply week1/homework/exercises/3-web-server/server.js]: True! I forgot to update that one as well. Hehe

**Comment by callejas1** [comment]
@olcarmontzaragoza thanks a lot for your taking your time to review my homework. I'm so glad to have refactored the function and IDK why I didn't think of the switch statement. What an on-point solution!
